x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/enrollment/RestCreateEnrollmentTokenAction.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.rest.action.enrollment;
+
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.action.RestBuilderListener;
+import org.elasticsearch.xpack.core.security.action.enrollment.CreateEnrollmentTokenAction;
+import org.elasticsearch.xpack.core.security.action.enrollment.CreateEnrollmentTokenRequest;
+import org.elasticsearch.xpack.core.security.action.enrollment.CreateEnrollmentTokenResponse;
+import org.elasticsearch.xpack.security.rest.action.SecurityBaseRestHandler;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+import static org.elasticsearch.rest.RestRequest.Method.PUT;
+
+/**
+ * Rest endpoint to create an enrollment token
+ */
+public class RestCreateEnrollmentTokenAction extends SecurityBaseRestHandler {
+    private final Settings settings;
+
+    /**
+     * @param settings the node's settings
+     * @param licenseState the license state that will be used to determine if
+     * security is licensed
+     */
+    public RestCreateEnrollmentTokenAction(Settings settings, XPackLicenseState licenseState) {
+        super(settings, licenseState);
+        this.settings = settings;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(
+            new Route(POST, "/_security/enrollment_token"),
```

Comment:
I know this is what the design doc proposes, but I think it's a mistake.
I think all the enrollment APIs should be under a common URI path.

That is the doc proposes:
```
GET /_security/enroll_node
POST /_security/enroll_client
POST /_security/enrollment_token
```

Putting aside the HTTP Method question, think the lack of a common prefix is a problem.
I would propose that everything is under `/_security/enroll/` or `/_security/enrollment/`

e.g.
```
POST /_security/enroll/node
POST /_security/enroll/client
POST /_security/enroll/token
```

Reply:
❤️ `/_security/enroll`~ment~